### PR TITLE
Use StackedArea-style tooltip in StackedBar

### LINKED ISSUE #1348: Use StackedArea-style tooltip in StackedBar
## Problem

In [stacked bars over time](https://ourworldindata.org/grapher/deaths-in-conflicts-and-one-sided-violence), we have a tooltip per each section of the bar:

https://user-images.githubusercontent.com/1308115/160829446-4c635ab8-2db1-4525-bb3f-98ab3d12c86f.mp4

They are often hard to hover over because they are so small, and the design of the tooltip isn't great (a lot of whitespace).

## Proposed solution

We should use the StackedDiscreteBar-style or StackedArea-style tooltip:

Screenshot 2022-03-30 at 13 00 54

### PR DESCRIPTION
**Pull request by MahmoudHamdy02**
Closes #1348 

This is a mostly finished draft for this issue, the new tooltip matches the style of the discrete bar chart. Please let me know what could be improved!

Current tooltip:
image

### CODE REVIEW

File: grapher/stackedCharts/StackedBarChart.tsx [hunk 1]
```diff
@@ -262,7 +262,7 @@ export class StackedBarChart
                                 </h3>
                             </td>
                         </tr>
-                        {series.map((series) => {
+                        {[...series].reverse().map((series) => {
```
Comment (MahmoudHamdy02) [inline grapher/stackedCharts/StackedBarChart.tsx:265]: Both `[...series].reverse()` and `series.slice().reverse()` work correctly, but I'm not sure if one of them is better.

Comment (danielgavrilov) [reply grapher/stackedCharts/StackedBarChart.tsx:265]: 👌 Either is fine

**Comment by danielgavrilov** [comment]
Thanks @MahmoudHamdy02. Sorry this took such a long time for me to review. I now made some design tweaks, but the rest is good!

@marcelgerber what do you think about this way of implementing the tooltip (like StackedArea) vs the way it's done on StackedDiscreteBar (with tippy)? I have a feeling the latter might work better here mostly due to the better positioning of tippy... but this is still an improvement over what we had before, so might be worth deploying without investing more time on this?

https://user-images.githubusercontent.com/1308115/171614766-0901542d-a583-4cd5-8420-9d2a432d1786.mp4

**Review by marcelgerber — COMMENTED** [review]
I totally agree @danielgavrilov!
It's a great improvement already, and even without Tippy we should just go ahead already.

It would also be nice not to duplicate the tooltip code into another file, but I that abstracting the tooltip is not super easy. So that's also fine for now.

**Review by marcelgerber — APPROVED** [review]

**Comment by marcelgerber** [comment]
Thank you @MahmoudHamdy02!

**Pull request merged**